features/nanostack/mbed-mesh-api/mbed-mesh-api/WisunInterface.h

```diff
@@ -185,24 +185,40 @@ class WisunInterface : public MeshInterfaceNanostack {
     /**
      * \brief Set Wi-SUN network size.
      *
-     * Function stores new parameters to mbed-mesh-api and uses them when connect() is called next time.
-     * If device is already connected to the Wi-SUN network then device will restart network discovery after
-     * changing the network size.
+     * Function stores network size parameter to the mbed-mesh-api and uses it when connect() is called for the next
+     * time. If a device is already connected to the Wi-SUN network, then the device will restart network discovery
+     * after changing the network size.
+     *
+     * It is recommended to set the correct network size because the Wi-SUN network configuration parameters are
+     * adjusted based on the selected network size. A network configured for a small amount of devices may not work
+     * optimally for large number of devices. This is because the network bandwidth is divided with all the devices in
+     * the network. Enough bandwidth must be reserved for application data usage as well as the Wi-SUN network
+     * operations. Application should adapt to the network characteristics by using the InternetSocket methods
+     * get_stagger_estimate_to_address and get_rtt_estimate_to_address.
+     *
+     * The network size is measured as hundreds of devices that are expected to join to the network. For example,
+     * for a 400-device network set network size to 4. The Wi-SUN stack will internally use one of the four possible
+     * configuration buckets based on the selected network size and data rate. The configuration buckets are defined
+     * for a small, medium, large and xlarge networks.
+     *
+     * For example, with 150kbs data rate set network size to:
+     * - 1 to use small configuration bucket.
+     * - 8 to use medium configuration bucket.
+     * - 15 to use large configuration bucket.
+     * - 25 to use xlarge configuration bucket.
```

Comment:
If we prefer customer to use the "number of hundreds devices" as network size, I would propose to remove all concept, micros and comments about " small, medium, large, xlarge",  the duplicate concept/usage will cause confusion。